platform/darwin/src/MGLOfflineStorage.mm

```diff
@@ -264,6 +264,62 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
     }
 }
 
+#pragma mark Offline merge methods
+
+- (void)addContentsOfFile:(NSString *)filePath withCompletionHandler:(MGLBatchedOfflinePackAdditionCompletionHandler)completion {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    if (![fileManager isWritableFileAtPath:filePath]) {
+        [NSException raise:NSInvalidArgumentException format:@"The file path must be writable"];
+    }
+    
+    __weak MGLOfflineStorage *weakSelf = self;
+    [self _addContentsOfFile:filePath withCompletionHandler:^(NSError * _Nullable error) {
+        if (error && completion) {
+            completion(error);
+        } else {
```

Comment:
If there’s an error but no completion block is provided, the code below invalidates all the packs, which seems wasteful. Perhaps this line should say `else if (!error)`?